### ISSUE TITLE
Matplotlib Example

### ISSUE THREAD

**Issue by wxprofessor** [matplotlib/basemap#389]
Hello! ... I reported the following issue in the Matplotlib repo... and they said to bring it to your attention:

The Matplotlib Sample: "Plot Precip With Filled Contours"
https://matplotlib.org/basemap/users/examples.html

code source: https://matplotlib.org/basemap/users/figures/plotprecip.py

seems to be dated... I say "seems" because this is my first week of python! :)

Code for reproduction
The variables as specified in your code do not seem to exist:
latcorners = nc.variables['lat'][:]
loncorners = -nc.variables['lon'][:]
lon_0 = -nc.variables['true_lon'].getValue()
lat_0 = nc.variables['true_lat'].getValue()

I'm guessing because that data is from 2006 and something has changed since then :)

I have been failing with new data located here:
http://water.weather.gov/precip/downloads/2018/01/24/nws_precip_1day_20180124_conus.nc

surely there is a way for matplotlib to cull the information needed from the file? if not I figure I'll do the long math..

use your code located here
code source: https://matplotlib.org/basemap/users/figures/plotprecip.py

but also notice that not only are the lat lon variables present, your variable:
prcpvar = nc.variables['amountofprecip']

will need to be adjusted to prcpvar = nc.variables['observation']

and the units are now inches instead of mm.

Here is the issue I submitted to matplotlib:
https://github.com/matplotlib/matplotlib/issues/10318

**Comment by gpotter2** [comment]
@dopplershift Hi ! I'm very new to this..

What is the status of the basemap project ? Is it dead ?
Should we use CartoPy instead or will it be still maintained ?

Thanks a lot for answering !

**Comment by WeatherGod** [comment]
I am the sole maintainer of basemap, and I am more here to make sure the
lights get turned off.

I am supporting basemap until 2020. This means that I'll fix things as I
come across them, and I'll review and merge pull requests that others make,
but I am not doing active development. Cartopy can do much of what basemap
could do, just differently. All new development should be done first with
Cartopy. If a feature is missing in cartopy that is in basemap, then it
should be considered a bug in Cartopy. Contributions and efforts should be
directed towards cartopy, not basemap.

**Comment by gpotter2** [comment]
@WeatherGod Thanks a lot for clarifying !

**Comment by jdkloe** [comment]
so wouldn't it be an idea to make cartopy a requirement for basemap and replace the basemap code behind the scenes with cartopy calls?

**Comment by WeatherGod** [comment]
I doubt that would be possible. The fundamental design of cartopy is very
different from basemap. Even the data that basemap works off of is
different from cartopy uses.

If you want to investigate that avenue, feel free to do so, but I have no
resources or inclination to do so myself.

**Comment by jdkloe** [comment]
Thanks, I understand the difficulty. Same for me, I don't have the possibility to spend a huge amount of time doing this.
Another idea is to provide some hints to the users. Maybe some of the example scripts could be modified to show how the same could be achieved using cartopy? With a little switch on top to choose the one or the other implementation?

**Comment by dopplershift** [comment]
I think the right answer is to get comparable examples at one of these two spots:

http://scitools.org.uk/cartopy/docs/v0.15/gallery.html
https://unidata.github.io/python-gallery

**Comment by efiring** [comment]
Right, cartopy badly needs attention, including additional examples.  Examples or tutorials along the lines of "how to translate this basemap plot into cartopy" would be enormously helpful.  As would clear identification of things one can easily do in basemap but not (yet) in cartopy.
